runtime/src/main/java/org/corfudb/runtime/Restore.java

```diff
@@ -136,23 +146,31 @@ private void enableCompaction() {
     private void restore() throws IOException {
         if (restoreMode == RestoreMode.FULL) {
             clearAllTables();
+        } else if (restoreMode == RestoreMode.PARTIAL_NAMESPACE) {
+            clearTablesInNamespace(namespace);
         }
 
         long startTime = System.currentTimeMillis();
         int restoredTableCount = 0;
-        for (String tableBackup : tableBackups) {
-            log.info("start restoring table {}", tableBackup);
-            if (restoreMode == RestoreMode.PARTIAL_TAGGED && !isTableTagged(tableBackup)) {
-                log.info("skip restoring table {} since it doesn't have requires_backup_support tag", tableBackup);
+        for (TableBackupFileName tableBackupFileName : tableBackupFileNames) {
+            log.info("start restoring table {}", tableBackupFileName);
+            if (restoreMode == RestoreMode.PARTIAL_TAGGED && !isTableTagged(tableBackupFileName)) {
+                log.info("skip restoring table {} since it doesn't have requires_backup_support tag", tableBackupFileName);
+                continue;
+            }
+
+            if (restoreMode == RestoreMode.PARTIAL_NAMESPACE &&
+                    !Objects.equals(tableBackupFileName.getNamespace(), namespace)) {
+                log.info("skip restoring table {} since it's not in namespace {}", tableBackupFileName, namespace);
                 continue;
             }
 
-            UUID streamId = UUID.fromString(tableBackup.substring(0, tableBackup.indexOf(".")));
+            UUID streamId = tableBackupFileName.getUuid();
             try {
-                Path tableBackupPath = Paths.get(restoreTempDirPath).resolve(tableBackup);
+                Path tableBackupPath = Paths.get(restoreTempDirPath).resolve(tableBackupFileName.toString());
                 restoreTable(tableBackupPath, streamId);
             } catch (IOException e) {
-                log.error("failed to restore table {} from temp file {}", streamId, tableBackup);
+                log.error("failed to restore table {} from temp file {}", streamId, tableBackupFileName);
                 throw e;
             }
             restoredTableCount++;
```

Comment:
nit: Can we also print totalNumTables in L179? Might be handy for debugging.
"successfully restored X out of Y tables"

Reply:
Done